### PR TITLE
Internal end-point for ejpd request processing.

### PR DESCRIPTION
**Pull request by fisx**
Attempt at streamlining the internal process at wire.com described [here](https://wearezeta.atlassian.net/wiki/spaces/~463749889/pages/256738296/EJPD+official+requests+process).

We are legally required to sometimes provide user information to law enforcement.  This change introduces an internal end-point to compile the minimum required information into a json blob.

**Security:** to access the internal end-point, you need the same access privileges that also give you direct access to the database.  So the information we provide here does not get any more accessible.  On the other hand, direct database access is error prone and make result in revealing information on the wrong user due to cut&paste errors.  This end-point mitigates this risk to a large extent.

### CODE REVIEW

**Comment by fisx** [comment]
```

POD LOGS: test-kwtrsypknule-wire-server-brig-integration

Parse failed: AesonException "Error in $: parsing Main.Config(Config) failed, key \"gundeck\" not found"

Falling back to environment variables

brig-integration: user error (failed to parse integration.yaml file)


Error: unable to get pod logs for test-kwtrsypknule-wire-server-cargohold-integration: pods "test-kwtrsypknule-wire-server-cargohold-integration" not found

make: *** [Makefile:262: kube-integration-test] Error 1
```

**Comment by fisx** [comment]
thanks @smatting: 3b8dfeb

File: libs/wire-api/src/Wire/API/Team/Member.hs [hunk 1]
```diff
@@ -195,10 +199,33 @@ type HardTruncationLimit = (2000 :: Nat)
 hardTruncationLimit :: Integral a => a
 hardTruncationLimit = fromIntegral $ natVal (Proxy @HardTruncationLimit)
 
+-- | Like 'ListType', but without backwards-compatible and boolean-blind json serialization.
+data NewListType
```
Comment (smatting) [inline libs/wire-api/src/Wire/API/Team/Member.hs:203]: Roundtrip test missing, I think

**Review by smatting — CHANGES_REQUESTED** [review]

File: libs/wire-api/src/Wire/API/Team/Member.hs (hunk 1 above)
Comment (fisx) [reply libs/wire-api/src/Wire/API/Team/Member.hs:203]: Hah, I think I was half-consciously hoping you'd miss that.  :)  My excuse is that it's included in the EJPD* roundtrip tests, but I'll add it anyway to protect against refactorings.

File: services/brig/src/Brig/Data/Connection.hs [hunk 2]
```diff
@@ -111,11 +112,15 @@ lookupConnectionStatus from to =
...
+lookupContactList u = fst <$$> lookupContactListWithRelation u
+
 -- | For a given user 'A', lookup the list of users that form his contact list,
 -- i.e. the users to whom 'A' has an outgoing 'Accepted' relation (A -> B).
-lookupContactList :: UserId -> AppIO [UserId]
-lookupContactList u =
-  map fst . filter ((== Accepted) . snd)
+lookupContactListWithRelation :: UserId -> AppIO [(UserId, Relation)]
+lookupContactListWithRelation u =
+  filter ((== Accepted) . snd)
```
[the remark this answers is not in the document]
Comment (fisx) [reply services/brig/src/Brig/Data/Connection.hs:123]: Even better catch!

**Comment by fisx** [comment]
```
    ejpd requests:                                                              FAIL
      Exception: ConnectionError (HttpExceptionRequest Request {
        host                 = "0.0.0.0"
        port                 = 8080
        secure               = False
        requestHeaders       = [("Accept","application/json;charset=utf-8,application/json"),("Content-Type","application/json;charset=utf-8")]
        path                 = "/i/ejpd-request"
        queryString          = ""
        method               = "POST"
        proxy                = Nothing
        rawBody              = False
        redirectCount        = 10
        responseTimeout      = ResponseTimeoutDefault
        requestVersion       = HTTP/1.1
      }
       (ConnectionFailure Network.Socket.connect: <socket: 46>: does not exist (Connection refused)))
```

Do we have to apply the change in 3b8dfeb manually?  But then why did the error change?

**Comment by fisx** [comment]
The host looks suspicious, I'm pretty sure `0.0.0.0` is not gundeck's true IP :).  Do I have to update something in the DNS that the brig-integration pod can see?

**Comment by fisx** [comment]
thanks @akshaymankar !  ef69c83

**Review by smatting — APPROVED** [review]

**Pull request merged**